src/coxph.jl

```diff
@@ -0,0 +1,85 @@
+function _cox_f(β, c::CoxAux{T})::T where T
+    grad = zeros(T, length(β))
+    hes = zeros(T, length(β), length(β))
+    _cox_fgh!(β, grad,hes, c::CoxAux{T}, false)
+end
+
+function _cox_fgh!(β, grad,hes, c::CoxAux{T}, compute_derivatives)::T where T
+    #get relevant quantities to compute negative loglikelihood, gradient and hessian
+
+    update_cox!(c, β, compute_derivatives)
+
+    X, ξ, Xβ, θ, Xθ, ξθ, λ, fs, ls  =
+        c.X, c.ξ, c.Xβ, c.θ, c.Xθ, c.ξθ, c.λ, c.fs, c.ls
+
+
+    #compute negative loglikelihood, gradient and hessian
+    y = zero(T)
+
+    if compute_derivatives
+        grad .= zero(T)
+        hes .= zero(T)
+
+        # preallocate
+        Z = zeros(T, length(β))
+    end
+
+    @inbounds for i in 1:length(fs)
+        for j in (fs[i]):(ls[i])
+            ρ = (j-fs[i])/(ls[i] + 1 - fs[i])
+            ϕ = θ.tails[i]-ρ * θ.chunks[i]
+            y -= Xβ[j] -log(ϕ)
+            if compute_derivatives
+                for k in eachindex(Z)
+                    Z[k] = Xθ.tails[i, k] - ρ * Xθ.chunks[i, k]
+                end
+                for k2 in 1:length(β)
+                    grad[k2] += Z[k2]/ϕ - X[j,k2]
+                    for k1 in 1:k2
+                        Ξ = ξθ.tails[i, k1, k2] - ρ * ξθ.chunks[i, k1, k2]
+                        hes[k1, k2] += Ξ/ϕ - Z[k1] * Z[k2]/ϕ^2
+                    end
+                end
+            end
+        end
+    end
+    y += λ*(β' * β)
+
+    if compute_derivatives
+        for k1 in 1:length(β)
+            grad[k1] +=  2*λ*β[k1]
+            hes[k1,k1] +=  2*λ
+        end
+        for k2 in 1:length(β)
+            for k1 in (k2+1):length(β)
+                hes[k1, k2] = hes[k2, k1]
+            end
+        end
+    end
+    return y
+end
+
+function _coxph(X::AbstractArray{T}, s::AbstractVector; l2_cost = zero(T), kwargs...) where T
+    c = CoxAux(X, s, l2_cost)
+
+    fgh! = (β,grad,hes, compute_derivatives) ->
```

Comment:
`_cox_fgh!` has one extra argument (the CoxAux type with all the model info). As I plan to use `newton_raphson` for other fits related to survival analysis, I prefer to keep it as a generic minimization routine.